### ISSUE TITLE
praznik filters applicability

### ISSUE THREAD

**Issue by bommert** [mlr-org/mlr#2786]
Why are only the praznik filters praznik_CMIM and praznik_MRMR applicable for regression data sets and the rest of the praznik filters can only be applied to classification data?
In the praznik package, it says for all of the filter functions that the target can be real-valued (but will be made categorical).

Also, I think that praznik_MRMR should not be allowed for survival data sets. I believe that it is conceptually not applicable and this error occurs when I apply the filter to a survival data set.
```
generateFilterValuesData(wpbc.task, "praznik_MRMR")
Error in .subset2(x, i, exact = exact) : subscript out of bounds
```

**Comment by pat-s** [comment]
@bommert It can very well be that some restrictions are not optimal. The implementation of praznik filters happened before my time so I cannot comment on it.

If you find some proof / reprex which shows that all these filters also work with regression datasets (and the internal conversion makes actually sense), I am happy to adjust the metadata.

The same goes applies for `praznik_MRMR` and survival datasets.

This would be a somewhat interesting task since it would also apply to {mlr3filters}. I pretty much copied the filter requirements from {mlr} when starting fresh.

**Comment by bommert** [comment]
In the documentation of the praznik package, it says:
"Real attributes are cut into about 10 equally-spaced bins, following the heuristic often used in literature. Precise number of cuts depends on the number of objects; namely, it is n/3, but never less than 2 and never more than 10."
"Attributes" refer to both the features and the target variable (they call it "decision attribute").

I think that different people may have different opinions on whether or not a mutual information based filter should be applied to regression data sets. On the one hand, this requires that numeric features and the target variable are categorized into a comparably small number of categories loosing information. On the other hand, we accept this loss of information for numeric features in classification data sets and if the categorization throws away too much information, we will observe this in the pretictive performance when the filter is combined with a predictive model.

With respect, to the "proof/reprex", what exactly would you like me to show you? The mlr praznik-filters (except CMIM and MRMR) currently do not work on regression data sets because of the choice of supported.tasks. Do you want to see an example of the praznik functions working with regression datasets without using mlr?

With respect to MRMR and survival data, the problem is that Y = data[[getTaskTargetNames(task)]] fails with the error I posted above. This is because survival tasks have two targets (usually a time and an indicator for censoring). Also, praznik expects a single target for its input Y.

**Comment by pat-s** [comment]
In a recent research project I've used the same binning approach for `FSelectorRcpp::information_gain()`.
I also did a small EDA comparing the effects of different binning sizes.

I am fine with opening up praznik filters like CMIM and friends to regression datasets following the binning approach.
Users should know what they are doing and whether this is statistically sound for them - not us :)

> With respect, to the "proof/reprex", what exactly would you like me to show you? 

A reprex showing tweaked mlr code which includes the necessary changes for opening praznik filters to regression datasets.
This could already be the content of the PR.

If you do not want to do the work twice, I'd suggest to contribute to {mlr3filters} instead of {mlr}.
Implementing something new in a deprecated package is not future-proof :)

> With respect to MRMR and survival data, the problem is that Y = data[[getTaskTargetNames(task)]] fails with the error I posted above. This is because survival tasks have two targets (usually a time and an indicator for censoring). Also, praznik expects a single target for its input Y.

Does this only apply to MRMR or to more filters? I am fine with removing the survival property if these filters are not applicable for survival tasks.
